server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -104,6 +92,29 @@ public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTyp
                 .getZonedDateTime());
     }
 
+    private Insert createSpecificUpdate(VacationUpdate vacationUpdate, Insert baseInsert) {
+        Insert insertWithSubject = addValue(baseInsert, CassandraVacationTable.SUBJECT, vacationUpdate.getSubject());
```

Comment:
The problem with your approch is that we have to reduce and build the pipeline only one time.

For this we need to comibine functions : 

```
@FunctionalInterface
    public interface CombinableFunction<A, R> extends Function<A, R> {
        default CombinableFunction<A, R> combine(CombinableFunction<A, R> other, BiFunction<R, R, R> resultCombiner) {
            return a -> {
                R r1 = apply(a);
                R r2 = other.apply(a);
                return resultCombiner.apply(r1, r2);
            };
        }
    }
```

Like this on each insert we just need to call : 

```
private Insert createSpecificUpdate(VacationPatch vacationPatch, Insert baseInsert) {
        return insertPipeline
            .apply(vacationPatch)
            .apply(baseInsert);
    }
```